src/components/TodoItem.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const [isChecked, setIsChecked] = useState(completed);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    setIsChecked(completed);
+  }, [completed]);
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newCompletedValue = event.target.checked;
+
+    setIsChecked(newCompletedValue);
+    const updatedTodo = { ...todo, completed: newCompletedValue };
+
+    const mapTodo = (prevTodos: Todo[]) => prevTodos.map(t => (t.id === id
+      ? updatedTodo : t));
+
+    setTodos(mapTodo(todos));
+  };
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleDeleteClick = (todoC: Todo) => {
+    const filterTodo = todos.filter((todoitem) => todoitem !== todoC);
+
+    setTodos([...filterTodo]);
```

Comment:
You're spreading the `filterTodo` array to create a new array, but `Array.filter` already returns a new array, so the spread operator is not necessary here.